src/containers/layout/chat-menu/ChatMenu.tsx

```diff
@@ -0,0 +1,153 @@
+import { FC, MouseEvent, useCallback } from 'react'
+import { useTranslation } from 'react-i18next'
+import { AxiosResponse } from 'axios'
+import MenuItem from '@mui/material/MenuItem'
+import UpdateDisabledIcon from '@mui/icons-material/UpdateDisabled'
+import DeleteOutlineIcon from '@mui/icons-material/DeleteOutline'
+
+import AppMenu from '~/components/app-menu/AppMenu'
+import useConfirm from '~/hooks/use-confirm'
+import useAxios from '~/hooks/use-axios'
+import { useSnackBarContext } from '~/context/snackbar-context'
+import { chatService } from '~/services/chat-service'
+
+import { styles } from '~/containers/layout/chat-menu/ChatMenu.styles'
+import { ChatResponse, ComponentEnum, ErrorResponse } from '~/types'
+import { defaultResponses, snackbarVariants } from '~/constants'
+
+interface ChatMenuProps {
+  anchorEl: Element | null
+  currentChat: ChatResponse
+  onClose: () => void
+  updateChats: () => Promise<void>
+}
+
+const ChatMenu: FC<ChatMenuProps> = ({
+  anchorEl,
+  currentChat,
+  onClose,
+  updateChats
+}) => {
+  const { t } = useTranslation()
+  const { openDialog } = useConfirm()
+  const { setAlert } = useSnackBarContext()
+
+  const onResponse = useCallback(() => {
+    setAlert({
+      severity: snackbarVariants.success,
+      message: 'chatPage.deleteSuccess'
+    })
+  }, [setAlert])
+
+  const onResponseError = useCallback(
+    (error: ErrorResponse) => {
+      setAlert({
+        severity: snackbarVariants.error,
+        message: error ? `errors.${error.code}` : ''
+      })
+    },
+    [setAlert]
+  )
+
+  const markAsDeletedService = useCallback(
+    (id?: string): Promise<AxiosResponse> =>
+      chatService.markChatAsDeleted(id ?? ''),
+    []
+  )
+
+  const deleteChatService = useCallback(
+    (id?: string): Promise<AxiosResponse> => chatService.deleteChat(id ?? ''),
+    []
+  )
+
+  const { fetchData: markAsDeleted } = useAxios({
+    service: markAsDeletedService,
+    defaultResponse: defaultResponses.object,
+    onResponse,
+    onResponseError,
+    fetchOnMount: false
+  })
+
+  const { fetchData: deleteChat } = useAxios({
+    service: deleteChatService,
+    defaultResponse: null,
+    onResponse,
+    onResponseError,
+    fetchOnMount: false
+  })
+
+  const handleDeletion = async (
+    id: string,
+    isConfirmed: boolean,
+    deletingFully: boolean
+  ) => {
+    if (isConfirmed) {
+      if (deletingFully) await deleteChat(id)
+      else await markAsDeleted(id)
```

Comment:
maybe better to use deletingFully ? await deleteChat : await markAsDeleted(id) ?

Reply:
> maybe better to use deletingFully ? await deleteChat : await markAsDeleted(id) ?

yeah, it would be much nicer to use just a ternary operator, but now I also added [message deletion](https://github.com/ita-social-projects/SpaceToStudy-Client/pull/1369/commits/801bf1c76564bfb74ff67669c4530ba9d0135426) before chat deletion, so, I can't use it for now 😅